### PR TITLE
Prevent package updates on packages with signature files

### PR DESCRIPTION
**Pull request by joelverhagen**
* Introduce a new configuration key `allowOverwritePackageWithSignatures`. By default it is `false`. When set to `true`, a replaced package has all of its pre-existing signature files removed before the new package is persisted. If `allowOverrideExistingPackageOnPush` is set to `false`, this new setting has no effect.
* Properly handle exceptions caused when you try to overwrite a package but cannot. A `400 Bad Request` is now returned.
* Clean up duplicate code in integration tests.

### CODE REVIEW

File: src/NuGet.Server/Web.config [hunk 1]
```diff
@@ -28,6 +28,14 @@
     <add key="allowOverrideExistingPackageOnPush" value="false" />
 
     <!--
+    Set allowOverwritePackageWithSignatures to false to prevent packages with signature files from beiing overwritten.
```
Comment (dtivel) [inline src/NuGet.Server/Web.config]: beiing => being

File: src/NuGet.Server/Web.config [hunk 2]
```diff
@@ -28,6 +28,14 @@
...
+    If this option is set to true, signature files associated with an overwritten package will be removed. This setting
```
Comment (dtivel) [inline src/NuGet.Server/Web.config]: This behavior isn't defined in the spec, but it probably should be.  The only reason to overwrite a package is to change it and that would invalidate signatures.

Comment (joelverhagen) [inline src/NuGet.Server/Web.config]: I'll document this in the spec.

File: src/NuGet.Server/Web.config (hunk 1 above)
Comment (joelverhagen) [inline src/NuGet.Server/Web.config]: Fixed.

File: src/NuGet.Server/Web.config (hunk 2 above)
Comment (joelverhagen) [inline src/NuGet.Server/Web.config]: Done.

**Pull request merged**